Sources/Puree/Output/BufferedOutput.swift

```diff
@@ -107,21 +112,30 @@ open class BufferedOutput: Output {
     @objc private func tick(_ timer: Timer) {
         if let lastFlushDate = lastFlushDate {
             if currentDate.timeIntervalSince(lastFlushDate) > flushInterval {
-                flush()
+                readWriteQueue.async {
+                    self.flush()
+                }
             }
         } else {
-            flush()
+            readWriteQueue.async {
+                self.flush()
+            }
         }
     }
 
     private func reloadLogStore() {
-        buffer.removeAll()
-
-        logStore.retrieveLogs(of: storageGroup) { logs in
-            buffer = buffer.union(logs)
+        readWriteQueue.sync {
+            buffer.removeAll()
+            let semaphore = DispatchSemaphore(value: 0)
```

Comment:
Do you need using semaphore?
I think you can use `DispatchGroup.notify`.

Reply:
`DispatchGroup.notify` cannot sync with readWriteQueue.